### PR TITLE
[Improve][Connector-V2] Imporve orc read strategy

### LINKED ISSUE #2713: [Improve][Orc-Read] Improve orc read strategy
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support read more complex data types from orc files. Now only support read basic types.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #2714: [Improve][Parquet-Read] Improve parquet read strategy
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support read more complex data types from parquet files. Now only support read basic types.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #2713

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by EricJoy2048 — DISMISSED** [review]
LGTM

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java [hunk 1]
```diff
@@ -163,6 +167,73 @@ boolean checkFileType(String path) {
...
     }
 
+    private SeaTunnelDataType<?> orcDataType2SeaTunnelDataType(TypeDescription typeDescription) {
+        switch (typeDescription.getCategory()) {
+            case BOOLEAN:
+                return BasicType.BOOLEAN_TYPE;
+            case INT:
+                return BasicType.INT_TYPE;
+            case BYTE:
+            case SHORT:
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java]: Why not use `TINYINT`、`SMALLINT`?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java [hunk 2]
```diff
@@ -163,6 +167,73 @@ boolean checkFileType(String path) {
...
+            case LONG:
+                return BasicType.LONG_TYPE;
+            case FLOAT:
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java:184]: Why not use :
```java
case FLOAT:
    return BasicType.FLOAT_TYPE
```

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java [hunk 3]
```diff
@@ -163,6 +167,73 @@ boolean checkFileType(String path) {
...
+            case DOUBLE:
+                return BasicType.DOUBLE_TYPE;
+            case STRING:
+            case BINARY:
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java:188]: Why not use :
```java
case BINARY:
    return PrimitiveByteArrayType.INSTANCE
```

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java [hunk 4]
```diff
@@ -163,6 +167,73 @@ boolean checkFileType(String path) {
...
+                    return ArrayType.INT_ARRAY_TYPE;
+                } else if (BasicType.LONG_TYPE.equals(seaTunnelDataType)) {
+                    return ArrayType.LONG_ARRAY_TYPE;
+                } else if (BasicType.FLOAT_TYPE.equals(seaTunnelDataType)) {
+                    return ArrayType.FLOAT_ARRAY_TYPE;
+                } else if (BasicType.DOUBLE_TYPE.equals(seaTunnelDataType)) {
+                    return ArrayType.DOUBLE_ARRAY_TYPE;
+                }
+                String errorMsg = String.format("SeaTunnel array type not supported this genericType [%s] yet", seaTunnelDataType);
+                throw new RuntimeException(errorMsg);
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java]: Use `switch (seaTunnelDataType.getSqlType())`?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java [hunk 5]
```diff
@@ -163,6 +167,73 @@ boolean checkFileType(String path) {
...
+                String[] fieldNames = typeDescription.getFieldNames().toArray(new String[0]);
+                SeaTunnelDataType<?>[] fieldTypes = children.stream().map(this::orcDataType2SeaTunnelDataType).toArray(SeaTunnelDataType<?>[]::new);
+                return new SeaTunnelRowType(fieldNames, fieldTypes);
+            case UNION:
+                // TODO: How to explain this type using SeaTunnel data type
+                throw new RuntimeException("SeaTunnel not supported orc [union] type yet");
+            default:
+                // do nothing
+                // never get in there
+                return null;
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java]: You will get unexpected results if the orc is upgraded, so log error message and throw exception

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java]: Becase these types in orc are saved in `LongVector`, so when we read these types using java api it is a `Long` object.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java (hunk 2 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/OrcReadStrategy.java:184]: Becase these types in orc are saved in `DoubleVector`, so when we read these types using java api it is a `Double` object.

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]
+1

**Comment by EricJoy2048** [comment]
@CalvinKirs  PTAL

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**